ui-ngx/src/app/modules/home/components/debug-settings/debug-settings-button.component.ts

```diff
@@ -21,82 +21,128 @@ import {
   ViewContainerRef,
   DestroyRef,
   ChangeDetectionStrategy,
-  EventEmitter,
-  Output
+  forwardRef
 } from '@angular/core';
 import { CommonModule } from '@angular/common';
 import { SharedModule } from '@shared/shared.module';
 import { DurationLeftPipe } from '@shared/pipe/duration-left.pipe';
 import { TbPopoverService } from '@shared/components/popover.service';
 import { MatButton } from '@angular/material/button';
-import { DebugConfigPanelComponent } from './debug-config-panel.component';
+import { DebugSettingsPanelComponent } from './debug-settings-panel.component';
 import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
 import { shareReplay, timer } from 'rxjs';
 import { SECOND } from '@shared/models/time/time.models';
-import { HasDebugConfig } from '@shared/models/entity.models';
+import { DebugSettings } from '@shared/models/entity.models';
 import { map } from 'rxjs/operators';
 import { getCurrentAuthState } from '@core/auth/auth.selectors';
 import { AppState } from '@core/core.state';
 import { Store } from '@ngrx/store';
+import {
+  ControlValueAccessor,
+  FormBuilder,
+  NG_VALUE_ACCESSOR,
+  UntypedFormGroup,
+} from '@angular/forms';
 
 @Component({
-  selector: 'tb-debug-config-button',
-  templateUrl: './debug-config-button.component.html',
-  styleUrls: ['./debug-config-button.component.scss'],
+  selector: 'tb-debug-settings-button',
+  templateUrl: './debug-settings-button.component.html',
   standalone: true,
   imports: [
     CommonModule,
     SharedModule,
     DurationLeftPipe,
   ],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => DebugSettingsButtonComponent),
+      multi: true
+    },
+  ],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class DebugConfigButtonComponent {
+export class DebugSettingsButtonComponent implements ControlValueAccessor {
 
-  @Input() debugFailures = false;
-  @Input() debugAll = false;
-  @Input() debugAllUntil = 0;
-  @Input() disabled = false;
-  @Input() minifyMode = false;
   @Input() debugLimitsConfiguration: string;
 
-  @Output() onDebugConfigChanged = new EventEmitter<HasDebugConfig>();
-
-  isDebugAllActive$ = timer(0, SECOND).pipe(map(() => this.debugAllUntil > new Date().getTime() || this.debugAll), shareReplay(1));
+  debugSettingsFormGroup: UntypedFormGroup;
+  disabled = false;
+  isDebugAllActive$ = timer(0, SECOND).pipe(map(() => this.allEnabledUntil > new Date().getTime() || this.allEnabled), shareReplay(1));
```

Comment:
For optimization, you don't need a timer for the value allEnabled to be true.  If the value is false, you need to stop the timer.

Reply:
it will overcomplicate the logic, in this way we need to make allEnabled observable and use switchMap, clearly not worth it